src/main/java/org/commcare/cases/query/queryset/DerivedCaseQueryLookup.java

```diff
@@ -0,0 +1,77 @@
+package org.commcare.cases.query.queryset;
```

Comment:
The tricky part about CaseIndexQueryTransform is that we can't put it in core, because core doesn't support the index table and the relevant operations needed to perform the transform. In core the only model query lookups that work are referential (IE:  anything that matches a case_id to a case_id).

I moved this here because it would let the bulk of the logic live in core for being re-implemented on other platforms. 

This code is triggered by  a test in the android codebase here:
https://github.com/dimagi/commcare-android/pull/1632/files#diff-c678f23327e97d00bb82c3717c08bbe0R105